common/src/main/java/io/netty/util/concurrent/LimitLatch.java

```diff
@@ -0,0 +1,111 @@
+package io.netty.util.concurrent;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.AbstractQueuedSynchronizer;
+
+/**
+ * Shared latch that allows the latch to be acquired a limited number of times
+ * after which all subsequent requests to acquire the latch will fail.
+ */
+public class LimitLatch {
+
+    private class Sync extends AbstractQueuedSynchronizer {

```

Comment:
> i made the assumption that we won't run task in boss loop.

This is not a good assumption. Tasks may be run on the boss loop, and the "boss loop" may be used to handle "worker channels" too.